examples/bench/bench_test.go

```diff
@@ -64,3 +68,118 @@ func BenchmarkAxon(b *testing.B) {
 		b.Errorf("average of CorSim for last %d epochs too low. Want %v, got %v", convergenceTestEpochs, 0.95, corSimAvg)
 	}
 }
+
+// Run just the threading benchmarks with `go test -bench=".*Thread.*" .`
+func benchmarkNeuronFunMultiThread(numThread int, b *testing.B) {
+	// this benchmark constructs the network just like `bench.go`, but without
+	// setting up the projections (not needed for benching NeuronFun) -> Test setup is much quicker.
+	units := 800 * 2048
+	threads := numThread
+
+	net := &axon.Network{}
+	net.InitName(net, "BenchNet")
+
+	squn := int(math.Sqrt(float64(units)))
+	shp := []int{squn, squn}
+
+	net.AddLayer("Input", shp, emer.Input)
+	net.AddLayer("Hidden1", shp, emer.Hidden)
+	net.AddLayer("Hidden2", shp, emer.Hidden)
+	net.AddLayer("Hidden3", shp, emer.Hidden)
+	net.AddLayer("Output", shp, emer.Target)
+
+	net.NThreads = threads // 1 overrides all
+
+	net.RecFunTimes = true
+
+	net.Defaults()
+	if _, err := net.ApplyParams(ParamSets[0].Sheets["Network"], false); err != nil {
+		panic(err)
+	}
+	// builds with default threads
+	if err := net.Build(); err != nil {
+		panic(err)
+	}
+
+	// override defaults: neurons, sendSpike, synCa, learn
+	net.Threads.Set(2, threads, threads, threads, threads)
+	net.ThreadsAlloc() // re-update
```

Comment:
Not clear what this comment means. I guess it should be unnecessary if the function has a good doc comment